sklearn/utils/_openmp_helpers.pyx

```diff
@@ -2,6 +2,11 @@ import os
 from joblib import cpu_count
 
 
+# Module level cache for cpu_count as we do not expect this to change during
+# the lifecycle of a Python program.
+_CPU_COUNTS = None
```

Comment:
instead of using ``globals``, you could directly make it a mutable
```suggestion
_CPU_COUNTS = dict()
```

Reply:
done